skorch/callbacks/lr_scheduler.py

```diff
@@ -149,12 +149,14 @@ def on_epoch_end(self, net, **kwargs):
             if callable(self.monitor):
                 score = self.monitor(net)
             else:
-                if self.lr_scheduler_.mode == 'max':
-                    score = -np.inf
-                elif self.lr_scheduler_.mode == 'min':
-                    score = np.inf
-                else:
+                try:
                     score = net.history[-1, self.monitor]
+                except KeyError as e:
+                    raise ValueError(
+                        f"'{self.monitor}' was not found in history. A "
+                        f"PassthroughScoring callback with name='{self.monitor}' "
+                        "should be placed before the LRScheduler callback"
+                    ) from e
 
             self.lr_scheduler_.step(score)
             # ReduceLROnPlateau does not expose the current lr so it can't be recorded
```

Comment:
Thanks for the fix in this PR, which I found while troubleshooting the same unexpected behavior as mentioned in the issue https://github.com/skorch-dev/skorch/issues/363. 

I noticed this comment but I guess how the learning rate was updated should be tracked in ``net.history`` even for `ReduceLROnPlateau`. As being discussed in https://discuss.pytorch.org/t/how-to-retrieve-learning-rate-from-reducelronplateau-scheduler/54234/3 and https://github.com/pytorch/pytorch/issues/2829, I would suggest replacing the comment line [162](https://github.com/skorch-dev/skorch/blob/cadb46914203cde74fcf08af0565cde8e059b0c0/skorch/callbacks/lr_scheduler.py#L162) with the following:

```
            if self.event_name is not None:
                net.history.record(self.event_name, self.lr_scheduler_._last_lr[0])
```

Reply:
We can discuss this in another PR since this is a feature unrelated to the bug fix.

Reply:
I created an issue (#739) so that we don't forget. @jiajiexiao does that reflect what you meant?

Reply:
Sure. Thanks, @thomasjpfan @BenjaminBossan